mapping/docs/user-guide/api-docs/openapi.yaml

```
@@ -0,0 +1,481 @@
+#  SPDX-FileCopyrightText: (C) 2025 Intel Corporation
+#  SPDX-License-Identifier: Apache-2.0
+
+openapi: 3.0.3
+info:
+  title: 3D Mapping Models API
+  description: |
+    REST API service for 3D reconstruction with build-time model selection.
+
+    This service provides endpoints for performing 3D reconstruction from multiple input images. The model is selected at build time:
+    - **MapAnything**: Universal Feed-Forward Metric 3D Reconstruction
+    - **VGGT**: Visual Geometry Grounded Transformer for sparse view reconstruction
+
+    Each container is built with a single model to avoid dependency conflicts and reduce image size.
+  version: 0.1.0
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+
+servers:
+  - url: http://localhost:8000
+    description: Local development server
+  - url: http://0.0.0.0:8000
+    description: Docker container server
```

Server URLs should be defined with a versioned base path

```suggestion
  - url: http://localhost:8000/v1
    description: Local development server
  - url: http://0.0.0.0:8000/v1
    description: Docker container server
```